sw/device/tests/clkmgr_off_peri_test.c

```diff
@@ -76,28 +161,87 @@ bool test_main(void) {
+      &flash_ctrl,
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+
   // Initialize aon timer.
   CHECK_DIF_OK(dif_aon_timer_init(
       mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
 
-  // Initialize uart0.
+  // Initialize peripherals.
   CHECK_DIF_OK(dif_uart_init(
       mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR), &uart0));
+  CHECK_DIF_OK(dif_spi_host_init(
+      mmio_region_from_addr(TOP_EARLGREY_SPI_HOST0_BASE_ADDR), &spi_host0));
+  CHECK_DIF_OK(dif_spi_host_init(
+      mmio_region_from_addr(TOP_EARLGREY_SPI_HOST1_BASE_ADDR), &spi_host1));
+  CHECK_DIF_OK(dif_usbdev_init(
+      mmio_region_from_addr(TOP_EARLGREY_USBDEV_BASE_ADDR), &usbdev));
+
+  // Enable cpu dump capture.
+  CHECK_DIF_OK(dif_rstmgr_cpu_info_set_enabled(&rstmgr, kDifToggleEnabled));
+
+  // Enable flash access
+  flash_ctrl_testutils_default_region_access(&flash_ctrl,
+                                             /*rd_en*/ true,
+                                             /*prog_en*/ true,
+                                             /*erase_en*/ true,
+                                             /*scramble_en*/ false,
+                                             /*ecc_en*/ false,
+                                             /*he_en*/ false);
 
   if (rstmgr_testutils_is_reset_info(&rstmgr, kDifRstmgrResetInfoPor)) {
-    // Enable watchdog bite reset.
-    CHECK_DIF_OK(dif_pwrmgr_set_request_sources(&pwrmgr, kDifPwrmgrReqTypeReset,
-                                                kDifPwrmgrResetRequestSourceTwo,
-                                                true));
     rstmgr_testutils_pre_reset(&rstmgr);
-    test_gateable_clocks_off(&clkmgr, 0);
+
+    // Starting clock.
+    dif_clkmgr_gateable_clock_t clock = kTopEarlgreyGateableClocksIoDiv4Peri;
+    LOG_INFO("Next clock to test %d", clock_to_test());
+
+    test_gateable_clocks_off(&clkmgr, &pwrmgr, clock);
 
     // This should never be reached.
     LOG_ERROR("This is unreachable since a reset should have been triggered");
     return false;
   } else if (rstmgr_testutils_is_reset_info(&rstmgr,
                                             kDifRstmgrResetInfoWatchdog)) {
-    return true;
+    dif_clkmgr_gateable_clock_t clock = clock_to_test();
+    LOG_INFO("Got an expected watchdog reset when reading for clock %d", clock);
+
+    size_t actual_size;
+    CHECK_DIF_OK(dif_rstmgr_cpu_info_get_size(&rstmgr, &actual_size));
+    // Verify the cpu crash dump.
+    dif_rstmgr_cpu_info_dump_segment_t cpu_dump[DIF_RSTMGR_CPU_INFO_MAX_SIZE];
+    size_t size_read;
+    CHECK_DIF_OK(dif_rstmgr_cpu_info_dump_read(
+        &rstmgr, cpu_dump, DIF_RSTMGR_CPU_INFO_MAX_SIZE, &size_read));
+    CHECK(size_read <= DIF_RSTMGR_CPU_INFO_MAX_SIZE);
+    CHECK(size_read == actual_size);
+    LOG_INFO("EXC_ADDR       = 0x%x", cpu_dump[0]);
+    LOG_INFO("EXC_PC         = 0x%x", cpu_dump[1]);
+    LOG_INFO("LAST_DATA ADDR = 0x%x", cpu_dump[2]);
```

Comment:
do we want to check that last_data_addr is the csr that should have hung? or is that checked somewhere else?

Reply:
lgtm.